tests/test_git_poller.py

```diff
@@ -139,3 +145,47 @@ def test_repo_changes_changed(self, get_refs):
             'GIT_REF': 'ref1',
         }]
         self.assertEqual(expected, list(change_params))
+
+    @mock.patch.object(git_poller.permissions, 'internal_post')
+    @mock.patch('jobserv.git_poller._github_log')
+    def test_skip_github(self, github_log, poster):
+        github_log.return_value = 'fake [skip ci] fake'
+
+        git_poller._trigger(
+            entry=_fake_entry(),
+            trigger_name='fake',
+            change_params={'GIT_URL': 'https://github.com'})
+
+        self.assertTrue(github_log.called)
+        self.assertIsNot(poster.called, True)
+
+    @mock.patch.object(git_poller.permissions, 'internal_post')
+    @mock.patch('jobserv.git_poller._gitlab_log')
+    def test_skip_gitlab(self, gitlab_log, poster):
+        gitlab_log.return_value = 'fake [ci skip] fake'
+        fake_gitlab_server = "https://fake.com"
+
+        with mock.patch(
+                'jobserv.git_poller.GITLAB_SERVERS',
+                [fake_gitlab_server]):
+            git_poller._trigger(
+                entry=_fake_entry(),
+                trigger_name='fake',
+                change_params={'GIT_URL': fake_gitlab_server},
+            )
+
+        self.assertTrue(gitlab_log.called)
+        self.assertIsNot(poster.called, True)
+
+    @mock.patch.object(git_poller.permissions, 'internal_post')
+    @mock.patch('jobserv.git_poller._cgit_log')
+    def test_skip_cgit(self, cgit_log, poster):
+        cgit_log.return_value = 'fake [skip ci] fake'
+
+        git_poller._trigger(
+            entry=_fake_entry(),
+            trigger_name='fake',
+            change_params={'GIT_URL': 'fake'})
+
+        self.assertTrue(cgit_log.called)
+        self.assertIsNot(poster.called, True)
```

Comment:
This is pretty good mocking. However, I think what you really want to test for here is that each _log function is checking for the skip messages properly as I noted above. So you probably want to directly call those functions and have mocks set up for the responses to what they call internally.

Reply:
Thank you Andy for this, I believe I've now approached your concerns. Please resolve or clarify if I've misunderstood.